### PR TITLE
fix(Core/Creatures): Renamed `CREATURE_FLAG_EXTRA_DONT_CALL_ASSISTANCE` to `CREATURE_FLAG_EXTRA_IGNORE_ASSISTANCE_CALL`

### LINKED ISSUE #12462: [ZG] 2 mob packs near Hakkar causing Hakkar chain pool if these pack are at Hakkar plane
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [x] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

2 mob packs near Hakkar causing Hakkar chain pool if these pack are at Hakkar plane. Screenshot with packs: https://ibb.co/ftHBrPV

### Expected Blizzlike Behaviour

These 2 packs should not chain pull Hakkar while pulling from stairs area of Hakkar plane (where boss is staying).

### Source

'---

### Steps to reproduce the problem

1. Go to ZG raid.
2. Go to Hakkar.
3. Aggro one pack from stairs lifting level to Hakkar level - Hakkar will pull and approach too in few seconds after the pull.

### Extra Notes

originally reported: https://github.com/chromiecraft/chromiecraft/issues/3797

Note seems to be a pathing related thing
**Note: Hakkar + two packs are sniffed.** 

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/35dfcf588f62339c44ff99a3e7e85f0b983ed923

### Operating system

Windows 10 Pro

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft Live - AzerothCore rev. 02fa20b6380b+ 2022-07-13 20:54:01 +0000 (master branch) (Unix, RelWithDebInfo, Static)

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #12462

## Issues Addressed:

- Closes #12462
- Closes (https://github.com/chromiecraft/chromiecraft/issues/3797)

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Go to ZG raid.
Go to Hakkar.
Aggro one pack from stairs lifting level to Hakkar level - Hakkar will not pull.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Most trash packs in ZG aren't linked in proper formations and are using assist to be pulled together instead. Could this break some pulls where those creatures are in?

**Comment by Gultask** [comment]
Yes, that'll be the case. It'll especially break the packs near Mandokir since they are mainly composed of Gurubashi Blood Drinkers and Hakkari Priests, which are affected by this PR.

I wanted to go around doing formations + linked respawns after WotLK Classic at level 80 is released to be sure.

**Comment by acidmanifesto** [comment]
> Yes, that'll be the case. It'll especially break the packs near Mandokir since they are mainly composed of Gurubashi Blood Drinkers and Hakkari Priests, which are affected by this PR.
> 
> I wanted to go around doing formations + linked respawns after WotLK Classic at level 80 is released to be sure.

Would you say the issue solution is improve upon or made worse based on your assessment?

**Comment by Gultask** [comment]
Worse, I'd say.

**Comment by UltraNix** [comment]
@Gultask I'm gonna make a generic solution so bosses cannot be aggroed by call assistance. Is that seems logic?

**Comment by Gultask** [comment]
Personally I don't see any issues coming with that. Some bosses when pulled also pull adds that have not been killed but that should be done through creature_formations and not assistance I'd think. So yeah, I don't see an issue with it.

**Comment by UltraNix** [comment]
@Gultask Please, test it.

**Comment by Gultask** [comment]
Yes, it works. I personally see no issues that can arise from this that cannot be fixed with formations.

**Comment by Nyeriah** [comment]
Not sure if bosses should always ignore call for assist from trash (i.e the dwarf boss in the nexus)

**Comment by Gultask** [comment]
That's true, it's not generic, I'm sorry. Checking in Retail right now, both Kurinnaxx and Moam from AQ20 can assist trash mobs.

**Comment by UltraNix** [comment]
@Gultask Please, retest it.

**Comment by Gultask** [comment]
Will revert PRs https://github.com/azerothcore/azerothcore-wotlk/pull/13041 and https://github.com/azerothcore/azerothcore-wotlk/pull/13042

**Comment by UltraNix** [comment]
It won't - this is just another way of working. Please, test it.

**Comment by Gultask** [comment]
Ok. Now it works fine 👍

**Review by Nefertumm — APPROVED** [review]

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -2423,6 +2423,12 @@ bool Creature::CanAssistTo(Unit const* u, Unit const* enemy, bool checkfaction /
     if (GetCharmerOrOwnerGUID())
         return false;
 
+    // Bosses cannot assist
```
Comment (Nefertumm) [inline src/server/game/Entities/Creature/Creature.cpp:2426]: Outdated comment?

Comment (Nefertumm) [inline src/server/game/Entities/Creature/Creature.cpp:2426]:
```suggestion
    // Check for ignore assistance extra flag
```

**Pull request merged**